Lib/site.py

```diff
@@ -509,35 +509,35 @@ def register_readline():
         pass
 
     if readline.get_current_history_length() == 0:
+        try:
+            from _pyrepl.main import CAN_USE_PYREPL
+        except ImportError:
+            CAN_USE_PYREPL = False
         # If no history was loaded, default to .python_history,
         # or PYTHON_HISTORY.
         # The guard is necessary to avoid doubling history size at
         # each interpreter exit when readline was already configured
         # through a PYTHONSTARTUP hook, see:
         # http://bugs.python.org/issue5845#msg198636
         history = gethistoryfile()
+        if os.getenv("PYTHON_BASIC_REPL") or not CAN_USE_PYREPL:
+            my_readline = readline
+        else:
+            my_readline = _pyrepl.readline
         try:
-            if os.getenv("PYTHON_BASIC_REPL"):
-                readline.read_history_file(history)
-            else:
-                _pyrepl.readline.read_history_file(history)
+            my_readline.read_history_file(history)
         except (OSError,* _pyrepl.unix_console._error):
             pass
 
         def write_history():
             try:
-                from _pyrepl.main import CAN_USE_PYREPL
-            except ImportError:
-                CAN_USE_PYREPL = False
-
-            try:
-                if os.getenv("PYTHON_BASIC_REPL") or not CAN_USE_PYREPL:
-                    readline.write_history_file(history)
-                else:
-                    _pyrepl.readline.write_history_file(history)
-            except (FileNotFoundError, PermissionError):
+                if my_readline.get_current_history_length() > 0:
```

Comment:
This adds a bit safety, but should we do this?

User should call ``_pyrepl.readline.clear_history()`` to wipe out history.   Maybe if he did this - it was on purpose and we should truncate the history file too?

Meanwhile, this condition wasn't tested.

Reply:
Do we assume `*.readline.clear_history()` is permanent?

Reply:
`readline.clear_history()` appears to be permanent.

Reply:
In this case, I belive that doing the check is nice, but breaks backward compatibility and thus it perhaps makes sense not to introduce it without a proper discussion.

Reply:
Yeah, I've added a suggestion in case reviewers would like to revert that part.